platform/storage/zephyr/nvs/src/mender-storage.c

```diff
@@ -207,6 +261,12 @@ mender_storage_get_deployment_data(char **deployment_data) {
         return ret;
     }
 
+#ifdef CONFIG_MENDER_STORAGE_DEPLOYMENT_DATA_CRC
+    if (MENDER_OK != check_crc(*deployment_data, deployment_data_length)) {
+        return MENDER_FAIL;
```

Comment:
What should we do if the crc is invalid?

As I read the code, what I see is that `mender_get_deployment_data` will return FAIL in `mender_client_initialization_work_function` and then go to REBOOT. It could be that we are in an state where reboot will bring us back to a working image (for example with MCU Boot, etc) but it could also be that it is not the case and we will keep rebooting forever.

I think that the safest is to erase the deployment data and let the deployment start again. Or? @vpodzime ?

What will happen to the state machine when we return `MENDER_FAIL` -> I suspect it will just go into the fail states path, but still won't be able to keep saving state and then the

Reply:
Also, related but clearly scope creep, this also applies if you are not able to read the private key from NVS at start up -> what should we do then?

(I am bringing this one in hoping to ease the discussion)

Reply:
That doesn't only happen if the CRC is invalid, though. It happens if the parsing fails or if the deployment data is invalid as well. We only call `mender_get_deployment_data` on a boot, right? So after a reboot on a deployment, if the deployment data for some reason is broken, isn't it perhaps safer to return back to the partition we started the deployment in?

Reply:
This is partially related to MEN-7555. I think we should log an error, reboot back into the working image and if the deployment data is not valid in there either, we should delete it. The biggest issue I see is that without deployment data we don't know for which deployment to log this error and report the failure status. So the deployment will be left in-progress on the server. If it **is** valid in the previous image, we should report a failure and the error.

Reply:
One note: remember that not always will be a second image, some Update Modules won't be "firmware update" but changing something in the current image, or in an external memory, or whatever that does not require reboot.

But okey, we can address this with MEN-7555 :+1: Thanks for the ideas